### ISSUE TITLE
MySQL and Postgres: Alert query syntax error

### ISSUE THREAD

**Issue by patricktokeeffe** [grafana/grafana#19063]
**What happened**:

Using this mysql database query (mostly autogenerated):
![image](https://user-images.githubusercontent.com/1348834/64816029-b8c26a80-d55b-11e9-86a7-000c0ccb8605.png)

...which creates this SQL...
```
SELECT
  UNIX_TIMESTAMP(time) DIV 300 * 300 AS "time",
  avg(batt_sens_V_Avg) AS "GZ1"
FROM geneseez1_tower_flux_notes
WHERE
  time BETWEEN FROM_UNIXTIME(1567712808) AND FROM_UNIXTIME(1568317608)
GROUP BY 1
ORDER BY UNIX_TIMESTAMP(time) DIV 300 * 300
```

...I get this nice graph:
![image](https://user-images.githubusercontent.com/1348834/64816644-eeb41e80-d55c-11e9-8e22-ecf000950958.png)

However, when I define an Alert
![image](https://user-images.githubusercontent.com/1348834/64816754-333fba00-d55d-11e9-9575-e816e3750cd8.png)

...and press "Test" it fails to run:
```
{
  "firing": true,
  "state": "pending",
  "conditionEvals": " = true",
  "timeMs": "-938.000ms",
  "error": "tsdb.HandleRequest() response error &{Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '<time_column>) as time_sec,\n  <value column> as value,\n  <series name column> as' at line 2  A 0xc0014856d0 [] []}",
  "logs": [
    {
      "message": "Condition[0]: Query",
      "data": {
        "from": 1568316311986,
        "queries": [
          {
            "refId": "A",
            "model": {
              "datasource": "TowerTest DB",
              "format": "time_series",
              "group": [
                {
                  "params": [
                    "$__interval",
                    "none"
                  ],
                  "type": "time"
                }
              ],
              "metricColumn": "none",
              "rawQuery": false,
              "rawSql": "SELECT\n  UNIX_TIMESTAMP(<time_column>) as time_sec,\n  <value column> as value,\n  <series name column> as metric\nFROM <table name>\nWHERE $__timeFilter(time_column)\nORDER BY <time_column> ASC\n",
              "refId": "A",
              "select": [
                [
                  {
                    "params": [
                      "batt_sens_V_Avg"
                    ],
                    "type": "column"
                  },
                  {
                    "params": [
                      "avg"
                    ],
                    "type": "aggregate"
                  },
                  {
                    "params": [
                      "GZ1"
                    ],
                    "type": "alias"
                  }
                ]
              ],
              "table": "geneseez1_tower_flux_notes",
              "timeColumn": "time",
              "timeColumnType": "datetime",
              "where": [
                {
                  "name": "$__timeFilter",
                  "params": [],
                  "type": "macro"
                }
              ]
            },
            "datasource": {
              "id": 2,
              "name": "TowerTest DB"
            },
            "maxDataPoints": 0,
            "intervalMs": 0
          }
        ],
        "to": 1568318111986
      }
    }
  ]
}
```

**What you expected to happen**:

Alert test should succeed. 

**How to reproduce it (as minimally and precisely as possible)**:

See above.

**Anything else we need to know?**:

Seems `$__timeFilter(time_column)` should have interpolation brackets: `$__timeFilter(<time_column>)`
```diff
               ],
               "metricColumn": "none",
               "rawQuery": false,
-              "rawSql": "SELECT\n  UNIX_TIMESTAMP(<time_column>) as time_sec,\n  <value column> as value,\n  <series name column> as metric\nFROM <table name>\nWHERE $__timeFilter(time_column)\nORDER BY <time_column> ASC\n",
+              "rawSql": "SELECT\n  UNIX_TIMESTAMP(<time_column>) as time_sec,\n  <value column> as value,\n  <series name column> as metric\nFROM <table name>\nWHERE $__timeFilter(<time_column>)\nORDER BY <time_column> ASC\n",
              "refId": "A",
```

**Environment**:
- Grafana version: 6.3.3 (84d0a71)
- Data source type & version: MySQL (built-in)
- OS Grafana is installed on: Windows 7 x64
- User OS & Browser: Windows 7 x64 + Chrome 76
- Grafana plugins: Carpet plot, Multistat, Windrose, DarkSky, GoogleCalendar, Sun and Moon
- Others: Nginx

**Comment by csheaupdesigns** [comment]
I am also experiencing this issue.  I posted to Grafana community and @patricktokeeffe pointed me to this issue.  My Grafana community post can be found at:

https://community.grafana.com/t/alert-rule-sql-syntax-error/20996.  

However, @patricktokeeffe 's description is more verbose and complete.

**Comment by patricktokeeffe** [comment]
fwiw, I've upgraded my instance to 6.3.5 and still obtain a mysql syntax error, even with totally different metric & alert queries. The mysql is MariaDB 10.3.17-winx64.

Seems that none of the query parameters are substituted into the raw sql.
> error": "tsdb.HandleRequest() response error &{Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '<time_column>) as time_sec,\n  <value column> as value,\n  <series name column> as' at line 2  A 0xc000874a20 [] []}
```
{
  "firing": true,
  "state": "pending",
  "conditionEvals": " = true",
  "timeMs": "36.000ms",
  "error": "tsdb.HandleRequest() response error &{Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '<time_column>) as time_sec,\n  <value column> as value,\n  <series name column> as' at line 2  A 0xc000874a20 [] []}",
  "logs": [
    {
      "message": "Condition[0]: Query",
      "data": {
        "from": 1568736047330,
        "queries": [
          {
            "refId": "A",
            "model": {
              "format": "time_series",
              "group": [
                {
                  "params": [
                    "$__interval",
                    "none"
                  ],
                  "type": "time"
                }
              ],
              "metricColumn": "none",
              "rawQuery": false,
              "rawSql": "SELECT\n  UNIX_TIMESTAMP(<time_column>) as time_sec,\n  <value column> as value,\n  <series name column> as metric\nFROM <table name>\nWHERE $__timeFilter(time_column)\nORDER BY <time_column> ASC\n",
              "refId": "A",
              "select": [
                [
                  {
                    "params": [
                      "PAR_density_Avg"
                    ],
                    "type": "column"
                  },
                  {
                    "params": [
                      "avg"
                    ],
                    "type": "aggregate"
                  },
                  {
                    "params": [
                      "PAR_density_Avg"
                    ],
                    "type": "alias"
                  }
                ]
              ],
              "table": "geneseez1_tower_flux",
              "timeColumn": "time",
              "timeColumnType": "datetime",
              "where": [
                {
                  "name": "$__timeFilter",
                  "params": [],
                  "type": "macro"
                }
              ]
            },
            "datasource": {
              "id": 2,
              "name": "Flux Towers Level 0"
            },
            "maxDataPoints": 0,
            "intervalMs": 0
          }
        ],
        "to": 1568736347330
      }
    }
  ]
}
```

**Comment by marefr** [comment]
Is this only when testing the alert rule? Does it resolve when saving the dashboard?

**Comment by patricktokeeffe** [comment]
After saving the dashboard and letting it fire, I get an alert email. It contains my alert body text plus the same syntax error message. Note the panel image comes through just fine.

> Error message
> --
> tsdb.HandleRequest() response error &{Error 1064: You have   an error in your SQL syntax; check the manual that corresponds to your   MariaDB server version for the right syntax to use near '<time_column>)   as time_sec, <value column> as value, <series name column> as' at   line 2 A 0xc000fc3550 [] []}

![image](https://user-images.githubusercontent.com/1348834/65064273-f268ec00-d934-11e9-9270-bdb4077384cb.png)

**Comment by marefr** [comment]
Was able to reproduce and have open a PR #19209. Interesting to know when this stopped working, but I guess it happen due to angular-reach migration.

**Comment by marefr** [comment]
Fix included in v6.4-beta2. Something to mention is that you need to edit an existing query having above problems and save the dashboard to resolve it. For new queries it should work out of the box as expected.
